app/src/org/commcare/activities/UpdateActivity.java

```diff
@@ -439,21 +439,27 @@ private static void reportAppUpdate() {
         DataChangeLogger.log(new DataChangeLog.CommCareAppUpdated());
     }
 
-    private void logoutOnSuccessfulUpdate() {
-        final String upgradeFinishedText =
-                Localization.get("updates.install.finished");
-        HiddenPreferences.setPostUpdateSyncNeeded(!isLocalUpdate && !BuildConfig.DEBUG);
-        HiddenPreferences.setShowXformUpdateInfo(true);
-        CommCareApplication.instance().expireUserSession();
+    private void returnResult() {
         if (proceedAutomatically) {
             finishWithResult(RefreshToLatestBuildActivity.UPDATE_SUCCESS);
         } else {
+            final String upgradeFinishedText = Localization.get("updates.install.finished");
             Toast.makeText(this, upgradeFinishedText, Toast.LENGTH_LONG).show();
             setResult(RESULT_OK);
             this.finish();
         }
     }
 
+    // Helper function for common operations to do after an app update
+    public static void OnSuccessfulUpdate(boolean logout, boolean syncPostUpdate){
```

Comment:
Moved the common after update operations in here. I am not sure whether `syncPostUpdate` should be `true` when we do the update as part of a recovery measure.

Reply:
I don't think it should matter too much whether we do the sync or not fornoiw